src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-report-modal.component.ts

```
@@ -1,151 +1,152 @@
-    private fetchSolutionRepoFiles() {
-        this.solutionRepoFilesSubscription = this.programmingExerciseService.getSolutionRepositoryTestFilesWithContent(this.report.programmingExercise.id!).subscribe({
-            next: (response: Map<string, string>) => {
-                this.rightCommitFileContentByPath = response;
-            },
-            error: () => {
-                this.errorWhileFetchingRepos = true;
-            },
-        });
+    private async fetchSolutionRepoFiles(): Promise<void> {
+        try {
+            const solutionRepoFiles = await firstValueFrom(this.programmingExerciseService.getSolutionRepositoryTestFilesWithContent(this.report().programmingExercise.id!));
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)